clients/src/main/java/org/apache/kafka/clients/admin/internals/DescribeConsumerGroupsHandler.java

```diff
@@ -103,6 +104,12 @@ public String apiName() {
         return new DescribeGroupsRequest.Builder(data);
     }
 
+    private void validateGroupsNotEmpty(List<DescribedGroup> describedGroups) {
+        if (describedGroups.isEmpty()) {
+            throw new InvalidGroupIdException("No consumer group found");
+        }
+    }
```

Comment:
Before KIP-699, there's the check:
```java
if (describedGroups.isEmpty()) {
          context.future().completeExceptionally(
                  new InvalidGroupIdException("No consumer group found for GroupId: " + context.groupId()));
          return;
      }
```
But, you're right, the exception thrown will not return back to requester. Also, because there's no group found, we cannot have a failed key return because we need a group in key. 

Removed it since it should never happen.